data_structures/hidden_markov_model.hpp

```diff
@@ -140,7 +140,7 @@ template <class CandidateLists> struct HiddenMarkovModel
             for (const auto s : osrm::irange<std::size_t>(0u, viterbi[initial_timestamp].size()))
             {
                 viterbi[initial_timestamp][s] =
-                    emission_log_probability(candidates_list[initial_timestamp][s].second);
+                    emission_log_probability(candidates_list[initial_timestamp][s].first);
```

Comment:
Out of scope. This PR is supposed to refactor the static rtree. If that's a HMM bug, handle separately.